x.zig

```diff
@@ -2478,8 +2541,171 @@ pub const ConnectSetup = struct {
     pub fn getScreensPtr(self: @This(), format_list_limit: u32) [*]align(4) Screen {
         return @alignCast(@ptrCast(self.buf.ptr + format_list_limit));
     }
+    pub fn getScreens(self: @This(), allocator: std.mem.Allocator) ![]align(4) *Screen {
+        const format_list_offset = ConnectSetup.getFormatListOffset(self.fixed().vendor_len);
+        const format_list_limit = ConnectSetup.getFormatListLimit(format_list_offset, self.fixed().format_count);
```

Comment:
The signatures of these functions are little bit cumbersome to use everywhere.

It seems like we could update them to just reference everything from `self` and handle the complexity of getting the right fields in the functions. Is there a benefit to the current way?

ex.

```zig
pub fn getFormatListOffset(self: @This) u32 {
    return VendorOffset + std.mem.alignForward(u32, self.fixed().vendor_len, 4);
}
pub fn getFormatListLimit(self: @This) u32 {
    return self.getFormatListOffset() + (@sizeOf(Format) * self.fixed().format_count);
}
```


Then when someone wants to get the format list, they can simply just call...

**Before:**
```zig
const format_list_offset = x.ConnectSetup.getFormatListOffset(fixed.vendor_len);
const format_list_limit = x.ConnectSetup.getFormatListLimit(format_list_offset, fixed.format_count);
const formats = try conn.setup.getFormatList(format_list_offset, format_list_limit);
```

**After:**
```zig
const formats = try conn.setup.getFormatList();
```

Reply:
(this is something for a future PR in any case)